server/students.js

```diff
@@ -58,7 +58,7 @@ router.post('/', function(req,res,next){
     const student = Student.build(req.body)
     student.setCampus(campus, { save: false})
     return student.save()
-      .then(studentOut => {
+      .then(studentOut => {//keep it flush!  Try your best not to nest then's unless absolutely necessary ***
```

Comment:
remember you should try to have all the .then's flushed, and not have nested promise chains.

Also, why not just res.json the student object?  Wouldn't the default scope and the include just send back the student and its associated campus automatically (after setting it)?